tests/test_api_client.py

```diff
@@ -0,0 +1,55 @@
+import sys
+import pytest
+import requests
+from pathlib import Path
+from unittest.mock import Mock, patch
+from dataclasses import dataclass
+
+sys.path.append(str(Path(__file__).resolve().parent.parent))
+from reaskapi.api_client import ClientConfig
+from reaskapi.deepcyc import DeepCyc
+from reaskapi.metryc import Metryc
+
+
+@dataclass
+class MockedResponse:
+    status_code: int = 200
+
+    def json(arg):
+        return {}
+
+
+@pytest.mark.parametrize("config", [None, ClientConfig("https://localhost:8001")])
```

Comment:
Should we try to avoid clashes on the 8001 port? e.g. if two people are testing on the same machine? If there is any easy way to deal with this otherwise we can leave it.

Reply:
The test here is just to confirm if `ClientConfig.base_url` is used when request.
I can set a random number of port number from pytests on `api-server` repo to avoid 'port-already-in-use' error.

Reply:
I would put `config_section` into the `ClientConfig` and create a new constructor which takes `ClientConfig` only. I leave the original constructor for backward compatibility but mark it as 'deprecated'.